src/renderer/components/ReactFlowBox.tsx

```diff
@@ -135,19 +147,69 @@ const updateZIndexes = (
         }
     }
 
+    const nodesToAdjust: { node: Node<NodeData>; zIndex: number }[] = [];
```

Comment:
Could you please explain all this zIndex logic? I don't get what this is supposed to do.